### DISCUSSION TITLE
Strange behaviour of simple Glide program

### DISCUSSION THREAD

**Discussion by Vort** [bochs-emu/Bochs#235] [Q&A]
I made simple program, which shows rotating triangle: [GlideTest_v1.zip](https://github.com/bochs-emu/Bochs/files/14003014/GlideTest_v1.zip).
When I launch it in Windows 95 with Voodoo 3, two strange things happen:
1. Image begin to flicker. Should it be considered as bug?
2. IPS drops to values always lower than specified in config. So it means CPU is overloaded regardless of how many instructions per second it needs to execute. How is that possible?

Here is the video of triangle rotation:

https://github.com/bochs-emu/Bochs/assets/1242858/9478459b-60cf-4e53-a368-e82ab32b9857

**Comment by stlintel** [comment]
The IPC drops that heavily usually means that some device model do heavy computations and not CPU model.
CPU model always runs ahead with instructions as each instruction is very simple.
CPU code physically cannot advance slower than average IPS without device model help :)

**Comment by stlintel** [comment]
Often these IPS drops happen due to I/O (log file is exploding with ERROR or INFO messages).
If not - might be good idea to use profiler to see where the time spent.

**Reply by Vort** [reply]
Probably my CPU is not fast enough to do 800x600@85Hz updates (why 85 Hz is another question, I'm setting `GR_REFRESH_60Hz` in `grSstWinOpen`), no matter what screen contents are. I will look at this problem in more detail, but for now here is profiler screenshot:
![image](https://github.com/bochs-emu/Bochs/assets/1242858/f2ed1a55-7788-4597-9dae-6dfe9a1caa10)

**Reply by Vort** [reply]
Profiler result is probably incomplete since I see lots of kernel mode load (red colored):
![image](https://github.com/bochs-emu/Bochs/assets/1242858/518526ef-f6a6-4ff2-9f20-9dfc865beffd)

**Reply by Vort** [reply]
If I change `update_freq=0, realtime=0` to `update_freq=30, realtime=0`, IPS no longer drops. Also kernel is less loaded:
![image](https://github.com/bochs-emu/Bochs/assets/1242858/a89fb4ca-a65b-46a2-bdc2-5ed0c0f92df6)
So question <span>#</span>2 have answer.
It is interesting to know what exactly consume resources, but that's another question.

Flickering is happening with 30 Hz update as well, so question <span>#</span>1 is still open.

**Comment by Vort** [comment]
Converted to issue #264.
